conans/client/profile_loader.py

```diff
@@ -238,6 +239,20 @@ def get_profile(profile_text, base_profile=None):
         platform_tool_requires = [RecipeReference.loads(r) for r in doc_platform_tool_requires.splitlines()]
         platform_requires = [RecipeReference.loads(r) for r in doc_platform_requires.splitlines()]
 
+        def load_replace(doc_replace_requires):
+            result = {}
+            for r in doc_replace_requires.splitlines():
+                r = r.strip()
+                if not r or r.startswith("#"):
+                    continue
+                src, target = r.split(":", 1)
+                target = RecipeReference.loads(target.strip())
```

Comment:
A `RecipeReference` performs no validation of its own, so there's nothing really stopping users from specifying a package id in the target (I think we break if specified as part of the src due to the split above), other than they getting a seemingly unrelated error like "Package 'foo/1.2:123' not resolved: No remote defined"

Do we want to guard against those cases? I wouldn't expect users to try to add those bits if we tell them it's not meant for things like these, but I never know anymore

Reply:
Yeah, better to add some checking and good error messages. I'll add it.